internal/pkg/secret/encryption_windows.go

```diff
@@ -0,0 +1,27 @@
+//go:build windows
+
+package secret
+
+import (
+	"github.com/billgraziano/dpapi"
+)
+
+func DeriveEncryptionKey(salt string) ([]byte, error) {
+	return []byte{}, nil
+}
+
+func Encrypt(password, _ string) (string, error) {
+	encryptedPassword, err := dpapi.Encrypt(password)
```

Comment:
Is this encryption associated with the logged in user or entire computer? DPAPI encrypts and associates data at two levels. 1. User and 2. Computer
If encryption is associated with Computer level, any user can decrypt the encrypted password. So it is best to encrypt data at User level.

Reply:
Both user id and machine id are being used to generate the encryption key. Do you think we should prompt user for system password too?

Reply:
@ankurjoshi9

Reply:
I don't think we generate encryption key in this case. This is DPAPI which automatically takes care of encryption.
I am trying to understand if we use https://github.com/billgraziano/dpapi/blob/master/dpapi.go#L116 method or not?

Reply:
No we don't use machine level encryption. so i think this is user level and we're fine